ttnn/cpp/ttnn/operations/normalization/batch_norm/device/batch_norm_program_factory.cpp

```diff
@@ -218,38 +221,80 @@ BatchNormOperation::BatchNormFactory::cached_program_t BatchNormOperation::Batch
         all_device_cores,
-        tt_metal::WriterDataMovementConfig({
-            b_is_dram,
-            c_is_dram,
-            d_is_dram,
-            e_is_dram,
-            f_is_dram,
-            static_cast<uint32_t>(weight_has_value),
-            static_cast<uint32_t>(bias_has_value),
-        }));
+        tt_metal::WriterDataMovementConfig(
+            {
+                b_is_dram,
+                c_is_dram,
+                d_is_dram,
+                e_is_dram,
+                f_is_dram,
+                static_cast<uint32_t>(weight_has_value),
+                static_cast<uint32_t>(bias_has_value),
+            },
+            std::move(writer_defines)));
 
     // COMPUTE KERNEL
     bool fp32_dest_acc_en = c_data_format == tt::DataFormat::UInt32 || c_data_format == tt::DataFormat::Int32 ||
                             c_data_format == tt::DataFormat::Float32;
+
+    uint32_t src_input_cb_index = tt::CBIndex::c_0;
+    uint32_t src_batch_mean_cb_index = tt::CBIndex::c_1;
+    uint32_t src_batch_var_cb_index = tt::CBIndex::c_3;
+    uint32_t src_eps_cb_index = tt::CBIndex::c_4;
+    uint32_t src_temp_den_cb_index = tt::CBIndex::c_5;
+    uint32_t src_temp_num_cb_index = tt::CBIndex::c_6;
+    uint32_t src_weight_cb_index = tt::CBIndex::c_16;
+    uint32_t src_temp_1_cb_index = tt::CBIndex::c_17;
+    uint32_t src_bias_cb_index = tt::CBIndex::c_18;
+
+    std::vector<UnpackToDestMode> unpack_to_dest_mode(NUM_CIRCULAR_BUFFERS, UnpackToDestMode::Default);
+    if (fp32_dest_acc_en) {
+        unpack_to_dest_mode[src_input_cb_index] = UnpackToDestMode::UnpackToDestFp32;
+        unpack_to_dest_mode[src_batch_mean_cb_index] = UnpackToDestMode::UnpackToDestFp32;
+        unpack_to_dest_mode[src_batch_var_cb_index] = UnpackToDestMode::UnpackToDestFp32;
+        unpack_to_dest_mode[src_temp_num_cb_index] = UnpackToDestMode::UnpackToDestFp32;
+        unpack_to_dest_mode[src_temp_den_cb_index] = UnpackToDestMode::UnpackToDestFp32;
+        unpack_to_dest_mode[src_eps_cb_index] = UnpackToDestMode::UnpackToDestFp32;
+        unpack_to_dest_mode[src_weight_cb_index] = UnpackToDestMode::UnpackToDestFp32;
+        unpack_to_dest_mode[src_temp_1_cb_index] = UnpackToDestMode::UnpackToDestFp32;
+        unpack_to_dest_mode[src_bias_cb_index] = UnpackToDestMode::UnpackToDestFp32;
+    }
+
     std::vector<uint32_t> compute_kernel_args = {
         static_cast<uint32_t>(weight_has_value), static_cast<uint32_t>(bias_has_value)};
     auto compute_kernel_id = tt_metal::CreateKernel(
         program,
-        "ttnn/cpp/ttnn/operations/normalization/batch_norm/device/kernels/compute/batch_norm_kernel.cpp",
+        fp32_dest_acc_en
+            ? "ttnn/cpp/ttnn/operations/normalization/batch_norm/device/kernels/compute/batch_norm_sfpu_kernel.cpp"
+            : "ttnn/cpp/ttnn/operations/normalization/batch_norm/device/kernels/compute/batch_norm_kernel.cpp",
```

Comment:
Maybe consider this?
```suggestion
        fmt::format("ttnn/cpp/ttnn/operations/normalization/batch_norm/device/kernels/compute/batch_norm_{}.cpp", fp32_dest_acc_en ? "sfpu_kernel" : "kernel"),
```